relecov_tools/assets/pipeline_utils/viralrecon.py

```
@@ -490,16 +490,27 @@ def quality_control_evaluation(data):
             qc_status = "pass"
             for param, condition in conditions.items():
                 value = sample.get(param)
-                if value is None or not condition(value):
+                try:
+                    if value is None or not condition(value):
+                        qc_status = "fail"
+                        break
+                except (TypeError, ValueError):
+                    if isinstance(value, str) and "Not Evaluable" in value:
+                        continue
                     qc_status = "fail"
+                    log_report.update_log_report(
+                        method_name,
+                        "warning",
+                        f"Sample {sample.get('sequencing_sample_id', 'unknown')} has unevaluable value for {param}: {value}",
```

I get your point here but sequencing_sample_id is mandatory so there's no way a sample does not have that field. And if the samples dont have that field, the logging would always show unknown, so you could not recognize the sample that failed anyway. I'd let it fail if sequencing_sample_id is not present or use sequence_file_R1 as you prefer